lib/prefab/config_resolver.rb

```diff
@@ -44,21 +43,22 @@ def raw(key)
       via_key ? via_key[:config] : nil
     end
 
-    def get(key, lookup_key, properties = {})
+    def get(key, properties = NO_DEFAULT_PROVIDED)
       @lock.with_read_lock do
         raw_config = raw(key)
 
         return nil unless raw_config
 
-        evaluate(raw(key), lookup_key, properties)
+        evaluate(raw(key), properties)
       end
     end
 
-    def evaluate(config, lookup_key, properties = {})
-      props = properties.merge(@additional_properties).merge(Prefab::CriteriaEvaluator::LOOKUP_KEY => lookup_key)
-
+    def evaluate(config, properties = NO_DEFAULT_PROVIDED)
       Prefab::CriteriaEvaluator.new(config,
-                                    project_env_id: @project_env_id, resolver: self, base_client: @base_client).evaluate(props)
+                                    project_env_id: @project_env_id,
+                                    resolver: self,
+                                    namespace: @base_client.options.namespace,
```

Comment:
Implementation detail I think, but I wonder whether thinking about this as 
```ruby
context = {user: { id: 455, name: "fred"}, 
                  prefab_client_options: { namespace: "foo" }}             
```
Would be less special case-y. Like we reserve a context for client options and set the property name to `prefab_client_options.namespace` in the UI.

Reply:
It would. I think I'll defer that until we have > 1 client option usage in this context. In the meantime, _not_ creating those objects is a small performance benefit.